juju/juju
# Pull in latest juju/os to support macOS Big Sur

Unfortunately the [juju/os](https://github.com/juju/os) repo's master branch had a breaking change applied to it, specifically the removal of `version.DefaultSupportedLTS()` (it moved into `juju/juju`). So updating required the same refactoring as we'd done on develop to use the `DefaultSupportedLTS` function in `juju/juju/version`.

We're pulling in these changes from `juju/os`, so the reviewer should also look here: https://github.com/juju/os/compare/8e6dd7a2b438d448578b32d0115ca0daaabb6d1f...10720519c304290ca6b6e7586a60ce0015323cfe -- I don't *think* the other changes should be a problem, as we're not changing the default series.

These changes were from https://github.com/juju/juju/pull/12103/files and https://github.com/juju/juju/pull/12108/files cherry-picked, with some manual fix-ups and a few conflict resolutions due to 2.8/develop differences.

## QA steps

Run `juju status` on macOS Big Sur and see if it works. I can't test as I don't have access to macOS Big Sur, but we'll ask the bug reporter to check. The relevant fix itself is a straight-forward one-liner in https://github.com/juju/os/commit/10720519c304290ca6b6e7586a60ce0015323cfe.

## Bug reference

https://bugs.launchpad.net/juju/+bug/1904252
